### PR TITLE
Revert "[TO REVERT] Never run TMVA GNN tests and tutorials as unit tests"

### PR DESCRIPTION
**Pull request by guitargeek**
This reverts commit baf092324e6e2a9301daff36a780cd064b232c2d.

We can either merge this as-is if the tests pass, or if they don't I'll close this PR without merging and open a GitHub issue about fixing these tests.

### CODE REVIEW

File: bindings/pyroot/pythonizations/test/CMakeLists.txt [hunk 1]
```diff
@@ -119,9 +119,7 @@ if (dataframe)
 endif()
 
 # SOFIE-GNN pythonizations
-# TODO: GNN tests are right now hardcoded to never run, because it is not clear
-# if they will pass on the new CI. Please revert this.
-if (FALSE AND tmva)
+if (tmva)
     if(NOT MSVC OR CMAKE_SIZEOF_VOID_P EQUAL 4 OR win_broken_tests)
         find_python_module(sonnet QUIET)
         find_python_module(graph_nets QUIET)
```
Comment (vepadulano) [inline bindings/pyroot/pythonizations/test/CMakeLists.txt:125]: Even re-enabling it on the CMakeLists still leaves this buggy way of finding the python module

**Comment by vepadulano** [comment]
I am a bit worried about seeing the CI fully green on this PR but at the same time the `sonnet` module was not even the right one to check for as highlighted by https://github.com/root-project/root/pull/15605

**Comment by guitargeek** [comment]
Yes, the results of this PR are not valid. We'll have to see again tomorrow when the correct sonnet versions makes it into the CI images. In fact, one can see in the CI logs that the GNN tests didn't run.

File: bindings/pyroot/pythonizations/test/CMakeLists.txt (hunk 1 above)
Comment (guitargeek) [reply bindings/pyroot/pythonizations/test/CMakeLists.txt:125]: What do you mean by that? I didn't experience this `find_python_module` function to be buggy

**Comment by guitargeek** [comment]
It worked! Here the relevant output from `alma9`:
```
2024-05-23T10:14:10.8127265Z  465/2558 Test   #62: pyunittests-pyroot-pyz-sofie-gnn ..............................................   Passed   16.85 sec
2024-05-23T10:19:21.5692440Z  922/2558 Test  #556: tutorial-tmva-TMVA_SOFIE_GNN_Parser ...........................................   Passed   27.96 sec
2024-05-23T10:21:56.4488110Z 1436/2558 Test #1092: tutorial-tmva-TMVA_SOFIE_GNN_Application ......................................   Passed   16.76 sec
2024-05-23T10:25:15.6946557Z 1723/2558 Test #1410: tutorial-tmva-TMVA_SOFIE_GNN-py ..............................................   Passed   26.54 sec
```

File: bindings/pyroot/pythonizations/test/CMakeLists.txt (hunk 1 above)
Comment (guitargeek) [reply bindings/pyroot/pythonizations/test/CMakeLists.txt:125]: Ah you probably mean that it makes it non-deterministic which tests are run

**Review by lmoneta — APPROVED** [review]
LGTM

**Pull request merged**